File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      String.last(input) == "?" -> "Sure."
      Regex.match?(~r/^ *$/, input) -> "Fine. Be that way!"
      Regex.match?(~r/\pL/, input) && String.upcase(input) == input -> "Whoa, chill out!"
      true -> "Whatever."
    end
  end
end

File: clojure/bob/test/bob.clj
(ns bob
  [:require [clojure.string :refer [ends-with? blank?]]])

(

(defn response-for [msg]
  (cond
    (blank? msg) "Fine. Be that way!"
    (let [letters (filter #(Character/isLetter %) msg)]
      (and (not (empty? letters))
           (every? #(Character/isUpperCase %) letters))) "Whoa, chill out!"
    (ends-with? msg "?") "Sure."
    :else "Whatever."))
